prow/github/types.go

```diff
@@ -1090,14 +1090,21 @@ type Milestone struct {
 // Note that it's wrapping a GitCommit, so author/committer information is in two places,
 // but contain different details about them: in RepositoryCommit "github details", in GitCommit - "git details".
 type RepositoryCommit struct {
-	SHA         string    `json:"sha"`
-	Commit      GitCommit `json:"commit"`
-	Author      User      `json:"author"`
-	Committer   User      `json:"committer"`
-	Parents     []Commit  `json:"parents,omitempty"`
-	HTMLURL     string    `json:"html_url"`
-	URL         string    `json:"url"`
-	CommentsURL string    `json:"comments_url"`
+	SHA         string         `json:"sha"`
+	Commit      GitCommit      `json:"commit"`
+	Author      User           `json:"author"`
+	Committer   User           `json:"committer"`
+	Parents     []ParentCommit `json:"parents,omitempty"`
+	HTMLURL     string         `json:"html_url"`
+	URL         string         `json:"url"`
+	CommentsURL string         `json:"comments_url"`
+}
+
+// ParentCommit represents the parent of commit.
+type ParentCommit struct {
```

Comment:
Can you copy the full typedef from https://github.com/google/go-github/blob/b47a3f75fc03ed477be4f04813180deb31407570/github/git_commits.go#L24 and make all scalar types literals and not pointers?

Reply:
I think this is unreasonable, because it has too many fields we don't need.

Reply:
And I think the structure of your link is not suitable for repo commit. You can check my previous [comment](https://github.com/kubernetes/test-infra/pull/19834#issuecomment-722966106).

Reply:
The [github docs](https://developer.github.com/v3/repos/commits/) mention all of the fields in that type, and its just as (un-)reasonable as your initial change, we don't use any of these fields

Reply:
Now the `RepositoryCommit` same as https://github.com/google/go-github/blob/b47a3f75fc03ed477be4f04813180deb31407570/github/repos_commits.go#L18 and `Commit` same as https://github.com/google/go-github/blob/b47a3f75fc03ed477be4f04813180deb31407570/github/git_commits.go#L24.